src/utils/verifyAuthSignedData.ts

```diff
@@ -0,0 +1,24 @@
+import { VerifyTypedDataParameters } from "viem";
+import { getEvmClient } from "./getRpcUrl.js";
+
+export const verifyAuthSignedData = async ({
```

Comment:
I'd be expecting tests here like:

- Cannot sign on wrong chain
- Cannot sign for other user
- Cannot submit message for other user
- Cannot submit for same user on different chain
- etc..

Reply:
Added some tests in https://github.com/hypercerts-org/hypercerts-api/pull/167/commits/d523e360bbff7d13afb741b376450af5fbce42f2, which seems to imply that specifying the domain in `verifyTypedData()` does get taken into account when verifying the signature. When we tested it before that didn't seem to happen but this makes a lot more sense. Will test from the UI to verify.

Reply:
Yes, this seems to be the case. Signing on a different chain errors out when using the chain_id that is stored in our db for the hyperboard/collection, as the chain_ids on the domain won't match.